app/models/assignment.rb

```diff
@@ -675,17 +675,23 @@ def summary_csv(role)
     if role.instructor?
       groupings = self.groupings
                       .includes(:group,
-                                :accepted_students,
                                 current_result: :marks)
+
     else
       groupings = self.groupings
                       .includes(:group,
-                                :accepted_students,
                                 current_result: :marks)
                       .joins(:memberships)
                       .where('memberships.role_id': role.id)
     end
 
+    students = Student.all
+                      .includes(accepted_groupings: :assignment)
+                      .where('accepted_groupings.assessment_id': self.id)
+                      .joins(:user)
+                      .order('users.user_name')
+                      .merge(groupings)
```

Comment:
This `merge` isn't necessary, likely the result of some earlier experimentation :)